src/components/Tooltip/Tooltip.js

```diff
@@ -46,35 +96,82 @@ class Tooltip extends Component {
             : tooltip;
     }
 
-    handleFocus = () => this.setState({ showTooltip: true })
+    getLeftPosition = position => ({
```

Comment:
Instead of build the object  every time the function is called, isn't it better to define it in a `const` and then just reuse it on each call?

Reply:
I don't think that this would be that good, because a component can move from a place to another depending on the logic of the siblings components. Let's imagine the scenario where you have two components one with the tooltip and another that is inserted into the DOM depending on a state boolean flag. So for the first time you open the page this flag is **false** and only the component with the tooltip is shown on the page. The user interacts and with the app and this flag is changed to **true** in some way, and then, we can see both components on the screen. Given that this function calculates the tooltip position it will be updated and we will be able to see a tooltip on the updated right place which we will probably don't reach if we set this constant.

Reply:
I see what you're saying, it might be slightly more performant to have a static object but I think that'll be tricky to do because it's using state values.

I don't think there will be enough of a performance problem from this to make it worthwhile changing. What do you think?

Reply:
Not a big deal, TBH. You'll need to have a function which would receive the state values and build the object anyway. Maybe this overhead of building the object could be avoided with a `switch () { case... }` statement.

Reply:
Yeah, it was a `switch` for a while but I thought this was a little clearer and cleaner to read. I'm pretty easy either way though.